pygtrie.py

```diff
@@ -1364,13 +1364,35 @@ def add(self, key):
             self._trie[key:] = True
 
     def discard(self, key):
-        raise NotImplementedError(
-            'Removing keys from PrefixSet is not implemented.')
+        try:
+            node, _ = self._get_node(key)
+        except:
+            pass
+
+        if node and node.children:
+            for child in node.children:
+                self.discard(child)
+        else:
+            node.children = {}
+            node.value = _SENTINEL
```

Comment:
This leaves out empty nodes which don’t lead to anything. For example, if you first add ``foo/bar/baz`` you get a prefix set with three nodes:

```
foo → bar → baz
```

Only ``baz``’s value is set. If you now remove ``foo/bar`` you end up with an empty set which nonetheless has two nodes:

```
foo → bar
```